rosapi/src/rosapi/params.py

```diff
@@ -93,11 +93,10 @@ def delete_param(name, params_glob):
         return
     # If the glob list is empty (i.e. false) or the parameter matches
     # one of the glob strings, continue to delete the parameter.
-    with param_server_lock:
-        if has_param(name):
+    if has_param(name, params_glob):
+        with param_server_lock:
```

Comment:
I double-checked the code if there is another place affected by this issue. I couldn't find any other places. Can you point out which code is possibly affected by this?